nearcore/src/config.rs

```diff
@@ -1097,32 +1099,68 @@ pub fn get_config_url(chain_id: &String) -> String {
     )
 }
 
-pub fn download_file(url: &String, path: &Path, limit: usize) {
-    actix::System::new().block_on(async move {
-        let client = awc::Client::new();
-        let mut response = client.get(url).send().await.expect("Unable to download the file");
-        // IMPORTANT: limit specifies the maximum size of the genesis or config file
-        // In case where the genesis or config file is bigger than the specified
-        // limit Overflow Error is thrown
-        let body = response
-            .body()
-            .limit(limit)
-            .await
-            .expect("File is bigger than specified limit. Please make the limit higher.");
-
-        std::fs::write(&path, &body).expect("Failed to create / write a file.");
+#[derive(thiserror::Error, Debug)]
+pub enum FileDownloadError {
+    #[error("Failed to download the file: {0}")]
+    HttpError(#[from] hyper::Error),
+    #[error("Failed to open file: {0}")]
+    OpenError(std::io::Error),
+    #[error("Failed to write to file: {0}")]
+    WriteError(std::io::Error),
+    #[error("Failed to rename file: {0}")]
+    RenameError(std::io::Error),
+    #[error("Invalid URI: {0}")]
+    UriError(#[from] hyper::http::uri::InvalidUri),
+    #[error("Failed to remove the temporary file after failure: {0}, {1}")]
+    RemoveTemporaryFileError(std::io::Error, Box<FileDownloadError>),
+}
+
+pub fn download_file(url: &String, path: &Path) -> Result<(), FileDownloadError> {
```

Comment:
Haha, I was following [this](https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#useless-types) style guide and was confused because I was sure that `&String` was the preferred method but apparently I misinterpeted the table :slightly_smiling_face: